Solution to various problems

presteamshop opened the pull request:
- fix issue #29 
- fix issue #38 
- fix issue #17 
- fix issue #35 
- fix issue #37 
- fix issue #39 (The "orderfees_payment" is now working only with redirection when using a one-page checkout module)

jimmyn reviewed (changes requested):
<img width="741" alt="image" src="https://github.com/user-attachments/assets/db5e75d2-b60f-474d-8490-495a757b4494">

The background of the card input should match. I'd make the font size and the padding match the default theme. Padding like on the main checkout button

<img width="1175" alt="image" src="https://github.com/user-attachments/assets/87560a51-caff-41ad-bb5c-52d690d52fdb">

Instead of the popup we should just show the spinner, check how this is implemented in Magento - https://magento2-demo.monei.com/

![2024-07-17 21 42 43](https://github.com/user-attachments/assets/6595b95d-fce5-4800-83a1-02582c852065)

Cardholder name field is required - https://docs.monei.com/docs/guides/send-cardholder-name/

<img width="1255" alt="image" src="https://github.com/user-attachments/assets/7d5ccfbd-8e8e-4dc0-94f9-ddf6c9d53de8">

I'm not redirected until I click OK, it should happen automatically, we should remove this popup completely and only show the Spiner. The error message should be shown as system error like this

<img width="1287" alt="image" src="https://github.com/user-attachments/assets/863d0115-838e-4e91-86fc-24ac864b7802">

Google Pay and Apple Pay should use Payment Request UI Component - https://docs.monei.com/docs/monei-js/payment-request/

presteamshop commented:
@jimmyn Just two comments based on what you mentioned:

1. The cardholder information is being sent with the logged-in customer's data, which is why I'm not requesting it separately.
2. Google and Apple Pay are complex developments, and I have decided not to proceed with them until the pending issues are resolved.

I will review the other points as soon as I can.

Best regards.

presteamshop commented:
@jimmyn I have done the following in a new commit:

1. The text field for entering the cardholder's name is now displayed.
2. I have removed the processing popup, leaving only the loading indicator.
3. I have removed the background from the card text box.
4. The automatic redirect section was already done; it just had not been updated on the online test store.
